File: C11/백준/Bronze/6888. Terms of Office/Terms of Office.c
#include <stdio.h>

int gcd(int a, int b) {
    while (b != 0) {
        int tmp = a % b;
        a = b;
        b = tmp;
    }
    return a;
}

int lcm(int a, int b) {
    return a * b / gcd(a, b);
}

int main() {
    int X, Y;
    scanf("%d", &X);
    scanf("%d", &Y);

    int interval = lcm(4, lcm(2, lcm(3, 5)));

    for (int year = X; year <= Y; year++) {
        if ((year - X) % interval == 0) {
            printf("All positions change in year %d\n", year);
        }
    }

    return 0;
}

File: Java/백준/Bronze/6888. Terms of Office/Terms of Office.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int X = Integer.parseInt(br.readLine());
        int Y = Integer.parseInt(br.readLine());

        int lcm = lcm(4, lcm(2, lcm(3, 5)));

        for (int year = X; year <= Y; year++) {
            if ((year - X) % lcm == 0) {
                System.out.println("All positions change in year " + year);
            }
        }
    }

    private static int gcd(int a, int b) {
        while (b != 0) {
            int tmp = a % b;
            a = b;
            b = tmp;
        }
        return a;
    }

    private static int lcm(int a, int b) {
        return a * b / gcd(a, b);
    }
}